torch/nn/functional.py

```diff
@@ -1307,11 +1307,7 @@ def gumbel_softmax(logits, tau=1, hard=False, eps=1e-10, dim=-1):
         https://arxiv.org/abs/1611.00712
         https://arxiv.org/abs/1611.01144
     """
-
-    if eps != 1e-10:
-        warnings.warn("`eps` parameter is deprecated and has no effect.")
-
-    gumbels = -torch.empty_like(logits).exponential_().log()  # ~Gumbel(0,1)
+    gumbels = -torch.empty_like(logits).exponential_().clamp_(min=eps).log()  # ~Gumbel(0,1)
```

Comment:
Rather than passing a hardcoded eps value, you could pass the following:
```
tiny = torch.finfo(logits.dtype).tiny
gumbels = -torch.empty_like(logits).exponential_().clamp_(min=tiny).log()
```

Reply:
Thanks for the suggestion. I have updated it.